src/Libraries/Microsoft.Extensions.AI/LoggingHelpers.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides internal helpers for implementing logging.</summary>
+internal static class LoggingHelpers
+{
+    /// <summary>Serializes <paramref name="value"/> as JSON for logging purposes.</summary>
+    public static string AsJson<T>(T value, JsonSerializerOptions? options = null)
+    {
+        if (options?.TryGetTypeInfo(typeof(T), out var typeInfo) is true ||
+            AIJsonUtilities.DefaultOptions.TryGetTypeInfo(typeof(T), out typeInfo))
+        {
+            return JsonSerializer.Serialize(value, typeInfo);
+        }
+
+        // Unable to get a type info for the value, so return an empty JSON object.
+        // We do not want lack of type info to disrupt application behavior with exceptions.
```

Comment:
Should we also eat any exceptions that may be thrown by the `JsonSerializer` call above?

Reply:
In what situations might it throw? The docs call out:
```
[ArgumentNullException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-8.0)
jsonTypeInfo is null.

[InvalidCastException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidcastexception?view=net-8.0)
value does not match the type of jsonTypeInfo.
```
neither of which should apply here. Are there other situations / exceptions? I'll change it to eat everything, that just always makes me slightly queasy.

Reply:
There are many reasons that it can throw relating to the shape of the underlying object graph. For example, there could be a nested `object` value that has an unsupported runtime type, or failing logic in a custom converter.